platform/tls/generic/mbedtls/src/mender-tls.c

```diff
@@ -447,7 +447,7 @@ mender_tls_user_provided_authentication_keys(mbedtls_pk_context *pk_context, con
 #if MBEDTLS_VERSION_NUMBER >= 0x03000000
     if (0 != (ret = mbedtls_pk_parse_key(pk_context, user_provided_key, user_provided_key_length, NULL, 0, mbedtls_ctr_drbg_random, ctr_drbg))) {
 #else
-    if (0 != (ret = mbedtls_pk_parse_key(pk_context, user_provided_key, user_provided_key_length, NULL, 0))) {
+    if (0 != (ret = mbedtls_pk_parse_key(pk_context, (const unsigned char *)user_provided_key, user_provided_key_length, NULL, 0))) {
```

Comment:
Please fix this also above. Latest Zephy LTS release comes with the most recent mbedtls version